From 30315f3fca4b922735b378d7cfb78b16c3b31569 Mon Sep 17 00:00:00 2001
From: Joan Torres <joan.torres@suse.com>
Date: Wed, 26 Feb 2025 18:05:04 +0100
Subject: [PATCH 3/7] wayland/color-management: Ensure ref lum can't be bigger
 than max lum

This might be possible if extended target volume is supported but
currently it isn't.

This is a requirement from the wayland color management:

"'reference_lum' may be higher than 'max_lum'. In that case reaching
the reference white output level in image content requires the
'extended_target_volume' feature support."

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4304>
---
 src/wayland/meta-wayland-color-management.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/wayland/meta-wayland-color-management.c b/src/wayland/meta-wayland-color-management.c
index 3c1a5fdf8..02f16edec 100644
--- a/src/wayland/meta-wayland-color-management.c
+++ b/src/wayland/meta-wayland-color-management.c
@@ -1137,6 +1137,15 @@ creator_params_set_luminance (struct wl_client   *client,
       return;
     }
 
+  if (ref > max)
+    {
+      wl_resource_post_error (resource,
+                              WP_IMAGE_DESCRIPTION_CREATOR_PARAMS_V1_ERROR_INVALID_LUMINANCE,
+                              "The reference luminance is bigger than the maximum luminance, "
+                              "extended target volume unsupported");
+      return;
+    }
+
   creator_params->lum.type = CLUTTER_LUMINANCE_TYPE_EXPLICIT;
   creator_params->lum.min = min;
   creator_params->lum.max = max;
-- 
2.45.3

